app/controllers/users.js

```diff
@@ -44,3 +45,34 @@ exports.getAllUsers = async (req, res, next) => {
     next(databaseError(DB_CONNECTION));
   }
 };
+
+exports.signUpAdmin = async (req, res, next) => {
+  const user = req.body;
+  const passEncrypted = await encryptPass(user.pass);
+  user.pass = passEncrypted;
+
+  try {
+    const userFound = await findUserByEmail(user.mail);
+    if (userFound.length) {
+      if (userFound[0].dataValues.role === ROLE.basic) {
+        logger.info(CHANGE_ROLE);
+        const dataToUpdate = { role: ROLE.admin };
+        const condition = { where: { id: userFound[0].dataValues.id } };
+        await updateUser(dataToUpdate, condition);
+        logger.info(ROLE_CHANGED);
+        return res.status(200).send({ message: ROLE_CHANGED });
+      } else if (userFound[0].dataValues.role === ROLE.admin) {
+        logger.info(USER_EXISTS);
+        return res.status(200).send({ message: USER_EXISTS });
+      }
+    }
+
+    if (!user.role) user.role = ROLE.admin;
+    await insertUser(user);
+    logger.info(`New user was created with role: ${ROLE.admin}`);
+    return res.status(201).send({ message: `New user was created with role: ${ROLE.admin}` });
+  } catch (e) {
+    logger.error(e);
+    return next(databaseError(DB_CONNECTION));
+  }
+};
```

Comment:
This is too long for a controller, let's abstract this in an interactor 😄 
The controller could be something like:
```js
exports.signUpAdmin = async (req, res, next) => {
  const user = req.body;
  try {
    await signUpAdminInteractor(user);
    logger.info(`New user was created with role: ${ROLE.admin}`);
    return res.status(201).send({ message: `New user was created with role: ${ROLE.admin}` });
  } catch (e) {
    logger.error(e);
    return next(databaseError(DB_CONNECTION));
  }
};
```